### ISSUE TITLE
⚠ Danger! Danger! envtpl is no longer maintained! ⚠

### ISSUE THREAD

**Issue by greg0ire** [TheLocehiliosan/yadm#159]
### This issue is about

* [ ] Man pages or command-line usage
* [ ] Website documentation
* [ ] Packaging
* [x] Other (dependencies)

### Describe the issue

I tried to use the jinja template functionality, and my first experience was a crash because of non-ASCII chars in the template. The thing is, I would very much like to use non-ASCII chars, so I looked at the stack trace and saw that the issue was in envtpl, one of your dependencies. I managed to fix the issue, and made [a PR](https://github.com/andreasjansson/envtpl/pull/31), only to find that [the author has dropped the ball on maintaining this lib](https://github.com/andreasjansson/envtpl/pull/29#pullrequestreview-214718692).

The choices are:

- do nothing and let all the bugs in envtpl affect yadm;
- drop the templating functionality;
- offer to maintain envtpl and get ownership on it;
- fork envtpl and maintain that fork;
- find another jinja2 processor in python.

What is it going to be?

**Comment by TheLocehiliosan** [comment]
Thanks for reporting that. I’ll look into adding support for j2cli.

**Comment by malikoth** [comment]
FWIW, I tried installing j2cli, and then I replaced line 227 of `yadm` with

```bash
j2 -e= "$tracked_file" > "$real_file"
```

And it worked flawlessly out of the box.

I tried putting together a PR that would  be backwards-compatible and use envtpl if it was available, and j2 otherwise if _it_ was available.  I couldn't get it working, because I'm no good at bash.  >< But `j2cli` really does work right out of the box, and it already contains the necessary encoding support to solve greg0ire 's problem with `envtpl`.  

We'd love to see you work your magic and get this in.  `j2cli` really seems much more well-thought-out and usable than `envtpl`, plus is still under active development.
